### PR TITLE
feat: implemented Enriched UI responses for StarSearch

### LINKED ISSUE #3372: Feature: Enriched Star Search Responses
As a user of StarSearch I want an enriched UI response that is more than standard HTML links, lists, etc. so that I can have more context in responses to my prompt.

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Implements an enriched UI for responses from StarSearch, e.g. adding a lottery factor chart when someone asks about lottery factor.

Currently the lottery factor widget is the only supported widget.

TODOS:

- [x] Now that things are working well for widgets loading in the UI, look at some refactoring (maybe)
- [x] Add Sentry logging where I put TODOs
- [x] Fix type for component loaded from dynamic import
- [x] Add some more comments in the code to explain the dynamic widget loading

## Related Tickets & Documents

Closes #3372

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/833231/91105855-b805-4953-a03c-9a18a083f9b5

## Steps to QA

1. Go to StarSearch
2. Ask a question along the lines of:
    - What is the lottery factor for React?
    - What is the lotto factor for React?
    - What is the bus factor for React?
3. The first response is a lottery factor chart.
4. The second response is a textual response in markdown converted to HTML.

Try existing canned prompts as well as your own to ensure things are still behaving properly.

## Tier (staff will fill in)

- [ ] Tier 1
- [x] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

File: components/StarSearchWidgets/LotteryFactorWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import LotteryFactorChart from "components/Repositories/LotteryFactorChart";
+import { useRepositoryLottoFactor } from "lib/hooks/api/useRepositoryLottoFactor";
```
Comment (nickytonline) [inline components/StarSearchWidgets/LotteryFactorWidget.tsx:2]: This is a thin wrapper component on the lottery factor chart that includes the same hook on the repo page.

This is probably the pattern we want to adopt for other widgets.

File: pages/star-search/index.tsx [hunk 2]
```diff
@@ -41,6 +51,28 @@ const SUGGESTIONS = [
 
 type SuggesionTypes = (typeof SUGGESTIONS)[number];
 
+async function updateComponentRegistry(name: string) {
```
Comment (nickytonline) [inline pages/star-search/index.tsx:54]: The amount of widgets dynamically loaded at the moment is low, one 😅 , but long term if the list of widgets explodes, an [LRU cache](https://redis.io/glossary/lru-cache/) might make sense especially in the context of mobile devices, but YAGNI for now.

**Review by brandonroberts — APPROVED** [review]

File: components/StarSearch/Widgets/LotteryFactorWidget.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+interface LotteryFactorWidgetProps {
+  repoName: string;
+}
+
+const LotteryFactorWidget = ({ repoName }: LotteryFactorWidgetProps) => {
+  const range = 30;
+  const { data, error, isLoading } = useRepositoryLottoFactor({ repository: repoName.toLowerCase(), range });
+
+  if (error) {
+    // whether it's a 404 or any other error, we don't want to rener the widget
```
Comment (nickytonline) [inline components/StarSearch/Widgets/LotteryFactorWidget.tsx:13]:
```suggestion
    // whether it's a 404 or any other error, we don't want to rener the widget
```
```suggestion
    // whether it's a 404 or any other error, we don't want to render the widget
```

**Review by zeucapua — APPROVED** [review]
emril-lagasse-gif

**Pull request merged**